source/music.cpp

```diff
@@ -0,0 +1,129 @@
+        /* NA_BGM_OCA_SARIA */          SEQ_OCARINA,
+        /* NA_BGM_OCA_EPONA */          SEQ_OCARINA,
+        /* NA_BGM_OCA_ZELDA */          SEQ_OCARINA,
+        /* NA_BGM_OCA_SUNMOON */        SEQ_OCARINA,
+        /* NA_BGM_OCA_TIME */           SEQ_OCARINA,
+        /* NA_BGM_OCA_STORM */          SEQ_OCARINA,
+        /* NA_BGM_NAVI */               SEQ_BGM,
+        /* NA_BGM_DEKUNOKI */           SEQ_BGM,
+        /* NA_BGM_FUSHA */              SEQ_BGM,
+        /* NA_BGM_HIRAL_DEMO */         SEQ_BGM,
+        /* NA_BGM_MINI_GAME */          SEQ_BGM,
+        /* NA_BGM_SEAK */               SEQ_BGM,
+        /* NA_BGM_ZORA */               SEQ_BGM,
+        /* NA_BGM_APPEAR */             SEQ_FANFARE,
+        /* NA_BGM_ADULT_LINK */         SEQ_BGM,
+        /* NA_BGM_MASTER_SWORD */       SEQ_FANFARE,
+        /* NA_BGM_INTRO_GANON */        SEQ_BGM,
+        /* NA_BGM_SHOP */               SEQ_BGM,
+        /* NA_BGM_KENJA */              SEQ_BGM,
+        /* NA_BGM_FILE_SELECT */        SEQ_NOSHUFFLE,
+        /* NA_BGM_ICE_DUNGEON */        SEQ_BGM,
+        /* NA_BGM_GATE_OPEN */          SEQ_NOSHUFFLE,
+        /* NA_BGM_OWL */                SEQ_BGM,
+        /* NA_BGM_DARKNESS_DUNGEON */   SEQ_BGM,
+        /* NA_BGM_AQUA_DUNGEON */       SEQ_BGM,
+        /* NA_BGM_BRIDGE */             SEQ_NOSHUFFLE,
+        /* NA_BGM_SARIA */              SEQ_NOSHUFFLE,
+        /* NA_BGM_GERUDO */             SEQ_BGM,
+        /* NA_BGM_DRUGSTORE */          SEQ_BGM,
+        /* NA_BGM_KOTAKE_KOUME */       SEQ_BGM,
+        /* NA_BGM_ESCAPE */             SEQ_BGM,
+        /* NA_BGM_UNDERGROUND */        SEQ_BGM,
+        /* NA_BGM_GANON_BATTLE_1 */     SEQ_BGM,
+        /* NA_BGM_GANON_BATTLE_2 */     SEQ_BGM,
+        /* NA_BGM_END_DEMO */           SEQ_NOSHUFFLE,
+        /* NA_BGM_STAFF_1 */            SEQ_NOSHUFFLE,
+        /* NA_BGM_STAFF_2 */            SEQ_NOSHUFFLE,
+        /* NA_BGM_STAFF_3 */            SEQ_NOSHUFFLE,
+        /* NA_BGM_STAFF_4 */            SEQ_NOSHUFFLE,
+        /* NA_BGM_BOSS01 */             SEQ_BGM,
+        /* NA_BGM_MINI_GAME_2 */        SEQ_BGM
+    };
+
+    std::array<u32, SEQ_COUNT> seqOverrides;
+
+    /* Initializes the list of music overrides to unshuffled */
+    void InitMusicRandomizer() {
+        for(int i = 0; i < SEQ_COUNT; i++)
+            seqOverrides[i] = BGM_BASE + i;
+    }
+
+    /* Shuffles the sequences grouping them by type */
+    /* type is a bitmask of SeqType */
+    void ShuffleSequences(int type) {
+        std::vector<u32> seqs;
+
+        // Get all sequences of the desired type(s) into a vector
+        for (int i = 0; i < SEQ_COUNT; i++) {
+            if (seqTypes[i] & type) {
+                seqs.push_back(seqOverrides[i]);
+            }
+        }
+
+        // Shuffle the vector...
+        for (std::size_t i = 0; i <= seqs.size(); i++)
+        {
+            std::swap(seqs[i], seqs[rand() %  (seqs.size() - i + 1) + i]);
```

Comment:
```suggestion
        for (std::size_t i = 0; i < seqs.size(); i++)
        {
            std::swap(seqs[i], seqs[rand() % seqs.size()]);
```
This fixes Ocarina Music sometimes becoming a fanfare when using these settings:
![image](https://user-images.githubusercontent.com/5352197/128317868-9883d810-7e51-4960-a048-256ce23d70a9.png)
I don't know if it fixes or breaks anything else.